### PR TITLE
Fix Redshift performance and memory issues for ingestion-base

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes:

We observed huge memory spikes when running the redshift ingestion, which came together with slow performance, when running the ingestion from the ingestion-base image.

image

- Changed a few things on the ingestion-base image, specifically to the postgres system libraries.
- After that, we also needed to fix the version sqlalchemy-redshift==0.8.12. Looks like from 0.8.12 to 0.8.13 something messed things up (we were using 0.8.14). Will look further on this one.

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/setup.py [hunk 1]
```diff
@@ -95,6 +95,7 @@ def get_long_description():
     "Jinja2>=2.11.3",
     "jsonpatch==1.32",
     "jsonschema",
+    "memory-profiler",
```
Comment (pmbrull) [inline ingestion/setup.py:98]: adding this by default, would help us debug similar scenarios later on with existing builds. It does not add any extra dependency

File: ingestion/operators/docker/Dockerfile [hunk 2]
```diff
@@ -29,23 +32,19 @@ RUN apt-get update \
...
-    wget --no-install-recommends
-
-# Prep to install msodbcsql18
-RUN apt-get update && \
-    apt-get install -y apt-transport-https && \
-    curl https://packages.microsoft.com/keys/microsoft.asc | apt-key add - && \
-    curl https://packages.microsoft.com/config/debian/11/prod.list > /etc/apt/sources.list.d/mssql-release.list && \
-    apt-get update && \
-    ACCEPT_EULA=Y apt-get install msodbcsql18 unixodbc-dev -y
+    vim \
```
[the remark this answers is not in the document]
Comment (pmbrull) [reply ingestion/operators/docker/Dockerfile:35]: I find it's usually a good-to-have, but can be removed

[the remark this answers is not in the document]
Comment (akash-jain-10) [reply ingestion/operators/docker/Dockerfile:35]: We can remove it for now! If needed for local testing, we can just keep it in `Dockerfile.ci` !

File: ingestion/setup.py [hunk 3]
```diff
@@ -213,7 +214,8 @@ def get_long_description():
     "redash": {VERSIONS["packaging"]},
     "redpanda": {*COMMONS["kafka"]},
     "redshift": {
-        "sqlalchemy-redshift~=0.8",
+        # Going higher has memory and performance issues
+        "sqlalchemy-redshift==0.8.12",
```
Comment (TeddyCr) [inline ingestion/setup.py:218]: @pmbrull did we open an issue on the `sqlalchemy-redshift` repo by any chance?

Comment (pmbrull) [reply ingestion/setup.py:218]: yeah on my todo list, let me do it right away

**Review by TeddyCr — APPROVED** [review]

**Review by akash-jain-10 — APPROVED** [review]

File: ingestion/setup.py (hunk 3 above)
Comment (pmbrull) [reply ingestion/setup.py:218]: https://github.com/sqlalchemy-redshift/sqlalchemy-redshift/issues/282

**Pull request merged**